python/nav/alertengine/base.py

```diff
@@ -457,26 +459,30 @@ def _verify_next_dispatch(queued_alert, now, _logger=_logger):
         # that the start period of the time period it was inserted in
         # has passed. This check should catch the corner case where
         # a user only has one timeperiod that loops.
-
-        if now.isoweekday() in [6, 7]:
-            valid_during = [TimePeriod.ALL_WEEK, TimePeriod.WEEKENDS]
-        else:
-            valid_during = [TimePeriod.ALL_WEEK, TimePeriod.WEEKDAYS]
-
-        only_one_time_period = active_profile.timeperiod_set.filter(
-            valid_during__in=valid_during).count() == 1
-
+        timeperiod_count = _get_number_of_timeperiods_today(active_profile, now)
         _logger.debug(
             'Tests: only one time period %s, insertion time %s',
-            only_one_time_period,
+            timeperiod_count == 1,
             insertion_time.time() < queued_alert_time_period.start)
 
         return is_different_timeperiod or (
-            only_one_time_period
+            timeperiod_count == 1
             and insertion_time.time() < queued_alert_time_period.start
         )
 
 
+def _get_number_of_timeperiods_today(alertprofile, now):
+    """
+    :type alertprofile: nav.models.profiles.AlertProfile
+    :type now: datetime
+    """
+    if now.isoweekday() in WEEKEND_DAYS:
```

Comment:
Exciting, since I started with the squashed view of all changes I knew that `WEEKEND_DAYS` was a thing. In the previous commit you didn't use it however and I wondered when it would turn up and now it did!

Reply:
Note that since you change this at the same time as moving the code block, there's no left-hand-side (minus-lines) in the diff.

(No, I will not start yakshaving on creating a better diff algorithm..)

Reply:
Haha, I could of course have done it in a separate commit, but I just couldn't resist placing some semantics into what I conceived as a bunch of magic numbers ;)